### ISSUE TITLE
JS SDK fails to initialize due to error: "Uncaught SecurityError: Failed to read the 'cookie' property from 'Document': The document is sandboxed and lacks the 'allow-same-origin' flag"

### ISSUE THREAD

**Issue by joyawad** [Microsoft/ApplicationInsights-JS#249]

**Comment by MaxShehovtsov** [comment]
Could you please describe your environment, is that a browser (if so which one), did you run it over http:// or file://.

Thanks!

**Comment by joyawad** [comment]
I have code that runs in an iframe (it's a VSTS widget) that is using the JS SDK. When loading the widget on a TFS on-prem dashboard, the JS SDK fails with the SecurityError. To access the on-prem dashboard, I go to http://<serverName>:8080/tfs/DefaultCollection/TestProject. It fails in IE 11 and Chrome 51.0.2704.106. I have a machine that repro's the issue if you need any more info. Thanks!

**Comment by dnduffy** [comment]
That URL is missing the host name.
Ignoring that for a moment, are you using a short name instead of an FQDN?

**Comment by joyawad** [comment]
The name is a short name (just the name of the machine). I tried with the full name (http://<machineName>.redmond.corp.microsoft.com:8080/tfs/DefaultCollection/TestProject)  and that appears to work fine (no exception thrown).

I have two customers reporting this issue so I can let them know about this workaround. 

Also, I have also been able to work around this issue by putting my trackEvent calls in a try/catch. The downside is my app won't send any AI telemetry when it's used with TFS on-prem and a short name.
